plutus-ledger-api/src/PlutusLedgerApi/V1/Value.hs

```diff
@@ -91,18 +102,26 @@ currencyMPSHash (CurrencySymbol h) = MintingPolicyHash h
 currencySymbol :: BS.ByteString -> CurrencySymbol
 currencySymbol = CurrencySymbol . PlutusTx.toBuiltin
 
--- | ByteString of a name of a token, shown as UTF-8 string when possible
+{- | ByteString of a name of a token, hashed with /BLAKE2b-256/.
+Shown as UTF-8 string when possible.
+Should be no longer than 32 bytes, empty for Ada.
+
+This is a simple type without any validation, __use with caution__.
+You may want to add checks for its invariants. See the
+ [Shelly ledger specification](https://hydra.iohk.io/build/16861845/download/1/ledger-spec.pdf).
```

Comment:
typo

Reply:
Maybe we should make clear that this has nothing to do with the mintingpolicyhash and is a non-unique arbitrary string attach to the currencysymbol/mintpolicyhash

Reply:
Or does it have to be unique in the AssetClass?

Reply:
An `AssetClass` is identified by a `CurrencySymbol` and a `TokenName` so I think `TokenName` can be anything for an `AssetClass`. A `Value` though, is a map from `CurrencySymbol`'s to a quantity of that currency, which is a map from `TokenName` to an `Integer`. So `CurrencySymbol`'s have to be unique and for each `CurrencySymbol` you need to have unique `TokenName`, otherwise there can be problems. But I think there is only one `TokenName` for each `CurrencySymbol` in an `AssetClass` so that should be fine. So the main thing is that `CurrencySymbol` have to be unique for the map to not lose info.

Reply:
I've updated the Haddock.

Reply:
I don't understand why `Value` isn't a map from `AssetClass` to `Integer` though.